share/getters/shrex.go

```diff
@@ -156,10 +157,13 @@ func (sg *ShrexGetter) GetEDS(ctx context.Context, root *share.Root) (*rsmt2d.Ex
 			return eds, nil
 		case errors.Is(getErr, context.DeadlineExceeded),
 			errors.Is(getErr, context.Canceled):
-			setStatus(peers.ResultCooldownPeer)
+			setStatus(peers.ResultRemovePeer)
 		case errors.Is(getErr, p2p.ErrNotFound):
 			getErr = share.ErrNotFound
 			setStatus(peers.ResultCooldownPeer)
+		case errors.Is(getErr, routing.ErrNotFound):
```

Comment:
We will need to remove this once https://github.com/celestiaorg/celestia-node/issues/2211 is fixed